scss/_patterns_search-box.scss

```diff
@@ -65,7 +65,7 @@
     border-left-style: solid;
     border-left-width: 1px;
     border-radius: 0 $border-radius $border-radius 0;
-    margin-right: $bar-thickness;
+    margin-right: $bar-thickness !important;
```

Comment:
I think the right fix here is to ensure we are wrapping all pattern elements with the block class as we agreed before.. 

For example: `.p-search-box .p-search-box__button`

This means you won't need the `!important`.